### PR TITLE
Scalar model pattern test cases

### PR DESCRIPTION
**Pull request by KanakaPathivada**
#### What type of PR is this?

- [ ] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

This PR continues expanding test coverage for scalar/constant arguments (following PR #1453), focusing on **dtype interactions** and **normalization operations** with scalar epsilon values. This addresses Issues #1074 and #1334 with comprehensive test coverage for production-critical scenarios.

**Key Changes:**
- **New test file:** `test_inductor_dtype_scalars.py` - Comprehensive dtype interaction testing
- **New test file:** `test_inductor_normalization_scalars.py` - Normalization operations with scalar parameters

**Test Coverage Added:**

### test_inductor_dtype_scalars.py (2 test classes, ~30+ test executions)

**TestDatatypeScalarOperations:**
- Mixed precision operations (FP64/FP32/FP16 combinations)
- NumPy scalar types (int8, int16, int32, uint8, float16/32/64)
- PyTorch 0-D scalar tensors with various dtypes
- Boolean scalar operations
- Type promotion scenarios (int8→FP32, FP16→FP64)
- Subnormal value handling
- Zero scalars (positive/negative)
- Empty tensors with scalar operations
- Large scalar values

**TestNegativeScalarOperations:**
- Invalid scalar type combinations
- Overflow/underflow scenarios
- NaN and Inf handling
- Device consistency validation

### test_inductor_normalization_scalars.py (2 test classes, ~30+ test executions)

**TestNormalizationScalarOperations:**
- **LayerNorm** patterns (with/without affine)
  - Various epsilon values (1e-5, 1e-6, 1e-8)
  - Multiple shapes and batch sizes
  - FP32 and FP16 dtypes
  
- **RMSNorm** (Root Mean Square Normalization)
  - Multiple epsilon configurations
  - With/without learnable weights
  
- **BatchNorm** (1D and 2D)
  - Identity running stats patterns
  - With affine transformations
  
- **GroupNorm**
  - Different group counts (8, 32)
  - With affine parameters
  
- **InstanceNorm**
  - 2D patterns with affine

**TestModelScalarOperations:**
- Transformer attention with scalar scaling
- Vision model normalization patterns
- Quantization with scalar factors

**Current Status:**
- ~60+ test executions across both files
- Most tests have xfail/skip markers with issue references
- All tests validate Spyre vs CPU behavior
- Comprehensive documentation of backend limitations

**Test Pattern:**
All tests use dual execution mode coverage (eager + compiled) and validate correctness via `compare_with_cpu()`.

### Test Execution Logs:

`pytest test_inductor_dtype_scalars.py`:
```
============================================================ test session starts ============================================================
platform linux -- Python 3.12.12, pytest-9.0.2, pluggy-1.6.0
rootdir: /home/mahalakshmip/torch-spyre-ext
configfile: pyproject.toml
collected 118 items                                                                                                                         

test_inductor_dtype_scalars.py xxxx...xxx.x.x.x.x..xx.x.x.xxx.xxxss.xxxxx..xx.x.x.x.x........ss..........ss..xxssxxxxxxxxxxxxxxssssss [ 86%]
....xx..xx...x..                                                                                                                      [100%]

============================================================= warnings summary ==============================================================
tests/inductor/test_inductor_dtype_scalars.py::TestNegativeScalarOperations::test_division_by_zero_result[eager]
  /home/mahalakshmip/torch-spyre-ext/tests/inductor/test_inductor_dtype_scalars.py:417: UserWarning: Backend Spyre does not support int64; downcasting to int32 may change values outside the 32-bit range. ....

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================== 51 passed, 14 skipped, 53 xfailed, 1 warning in 37.56s ===========================================
```
`pytest test_inductor_normalization_scalars.py`:
```
============================================================ test session starts ============================================================
platform linux -- Python 3.12.12, pytest-9.0.2, pluggy-1.6.0
rootdir: /home/mahalakshmip/torch-spyre-ext
configfile: pyproject.toml
collected 90 items                                                                                                                          

test_inductor_normalization_scalars.py xxxxxxxx..........xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.xx..........xx........xxxx..x...xx..     [100%]

====================================================== 36 passed, 54 xfailed in 58.38s ======================================================
```

#### Which issue(s) this PR is related to:

**Primary Issues:**
- #1074 - Increase test coverage for operators with scalar/constant as positional arguments
- #1334 - Constant tensor creation fails with IndexError during layout propagation

**Secondary Issues (Documented in Tests):**
- #1454 - to_dtype on float32 not supported / Mean reduction on float32 not supported
- #925 - int32/int16 dtype causes Signal Abort in data format converter
- #1487 - 0-dim tensor copy causes segmentation fault
- #1377 - Broadcasting size-1 dimensions - cannot map stick expr to host dimension
- #1531 - Square root operation on float32 (IEEE_FP32) not supported
- #1722 - view() + mean() triggers memory span limit (256MB) without splitting reduction dimensions
- #1558 - In-device copy not implemented (related to #1384/#1381)
- #1639 - dtype conversion not supported in add operations
- #1730 - Compilation issue for large sequence length (seq=1024)

**Additional Issues (Documented in Tests):**
- #1228 - FP64 (Double) dtype not supported
- #1545 - ComplexFloat (torch.complex64) dtype not supported
- #1588 - 128-byte alignment requirement for uint8 tensors
- #1688 - Variance (aten::var.correction) operation not implemented
- #1219 - aten::index operation not supported for NaN/Inf validation
- #1738 - aten::isnan operation not supported
- #1387 - Clamp (aten::clamp) operation not implemented
- #1723 - Round (aten::round.out) operation not implemented
- #1740 - UnimplementedOp missing iteration_space attribute
- #1598 - Device mismatch error in compiled mode
- #543 - aten::_reshape_alias not implemented (related to #1374)
- #1731 - Bilinear logits compilation failure
- #453 - LogSumExp eager mode issue

**Total Issues Addressed:** 24 unique issues (11 primary/secondary + 13 additional)

**Context:**
This PR builds upon PR #1453 (basic arithmetic and SDPA tests) by adding comprehensive dtype interaction and normalization operation coverage, addressing critical production use cases.

**Test Quality:**
- All test names accurately describe their logic
- Proper xfail/skip markers with issue references
- Both eager and compiled mode coverage
- CPU vs Spyre validation for correctness
- Appropriate tolerance values for each dtype (see tolerance reference below)

**Tolerance Values:**
Tests use dtype-appropriate tolerance values for CPU vs Spyre validation:
- **FP32 operations:** `atol=1e-4, rtol=1e-3` (standard precision)
- **FP16 operations:** `atol=1e-3, rtol=1e-2` (reduced precision)
- **FP64 operations:** `atol=1e-10, rtol=1e-7` (high precision)
- **Integer scalars:** `atol=0.004, rtol=0.01` (integer-to-float conversion)
- **Quantization ops:** `atol=0.1, rtol=1e-3` (relaxed for quantization error)
- **Attention patterns:** `atol=1e-1, rtol=1e-1` (relaxed for complexity)

These tolerance values are embedded directly in the test files and reflect the precision requirements for each operation type.

#### Does this PR introduce a user-facing change?

No. This is a test-only PR that expands test coverage for scalar/constant operations. It does not modify any production code or user-facing APIs.

The test suite is designed for incremental enablement - as each backend issue is resolved, corresponding tests can be enabled by removing skip markers.

### CODE REVIEW

**Review by dgrove-oss — APPROVED** [review]
lgtm

File: tests/inductor/test_inductor_dtype_scalars.py [hunk 1]
```diff
@@ -0,0 +1,767 @@
...
+
+@pytest.mark.filterwarnings("ignore::torch_spyre.ops.fallbacks.FallbackWarning")
+@pytest.mark.parametrize("execution_mode", ["eager", "compiled"])
+class TestDatatypeScalarOperations:
+    """
+    Scalar–tensor dtype interactions: promotions, NumPy/Python scalars, 0-D torch tensors,
+    shapes, and edge magnitudes. Spyre vs CPU via ``compare_with_cpu`` where applicable.
+    """
+
+    torch.manual_seed(0xAFFE)
```
Comment (dgrove-oss) [inline tests/inductor/test_inductor_dtype_scalars.py:49]: torch.manual_seed should  be in a setup method

File: tests/inductor/test_inductor_dtype_scalars.py [hunk 2]
```diff
@@ -0,0 +1,767 @@
...
+
+@pytest.mark.filterwarnings("ignore::torch_spyre.ops.fallbacks.FallbackWarning")
+@pytest.mark.parametrize("execution_mode", ["eager", "compiled"])
+class TestNegativeScalarOperations:
+    """
+    Errors, non-finite values, promotion rules, in-place and compile edge cases.
+    Uses ``_run_spyre`` when assertions are not expressible as CPU reference equality.
+    """
+
+    torch.manual_seed(0xAFFE)
```
Comment (dgrove-oss) [inline tests/inductor/test_inductor_dtype_scalars.py:375]: same here.  please move to a setup method

**Review by dgrove-oss — CHANGES_REQUESTED** [review]

File: tests/inductor/test_inductor_dtype_scalars.py (hunk 2 above)
Comment (dgrove-oss) [reply tests/inductor/test_inductor_dtype_scalars.py:375]: Actually, we make that mistake other places in the test suite.  Will do in a separate PR.

**Review by dgrove-oss — APPROVED** [review]
